cores/arduino/stm32/twi.c

```diff
@@ -338,7 +336,10 @@ i2c_status_e i2c_master_write(i2c_t *obj, uint8_t dev_address,
   HAL_StatusTypeDef status = HAL_OK;
 
   // Check the communication status
-  status = HAL_I2C_Master_Transmit(&(obj->handle), dev_address, data, size, I2C_TIMEOUT_TICK);
+  status = HAL_I2C_Master_Transmit_IT(&(obj->handle), dev_address, data, size);
+
+  // wait for transfer completion
+  while(HAL_I2C_GetState(&(obj->handle)) != HAL_I2C_STATE_READY){};
```

Comment:
Maybe add a timeout in case of error.

Reply:
Either check for a timeout indeed or maybe at least cehck for possible errors